app/helpers/application_helper.rb

```diff
@@ -568,46 +568,57 @@ def model_report_type(model)
     nil
   end
 
+  def show_taskbar_in_header?
+    return false if @explorer
+    return false if controller.action_name.end_with?("tagging_edit")
+
+    hide_actions = %w(
+      auth_error
+      change_tab
+      show
+    )
+    return false if @layout == "" && hide_actions.include?(controller.action_name)
+
+    hide_layouts = %w(
```

Comment:
@NickLaMuro : You are right. I was so happy with the logic flow cleanup that I did not even consider that. Also we do this in so many places that I've lost any sensitivity to it.

Also: the goal with these methods is to remove hard-coded lists and have instead properties on controllers and views or methods that would be queried.